### ISSUE TITLE
Metal slug xx

### ISSUE THREAD

**Issue by Davidcard22** [hrydgard/ppsspp#13916]
### What happens?
broken images small crashes almost imperceptible but harmful to gambling especially when there are many enemies on the screen this is problems with the way the  I / O is GPU is being used 

### What *should* happen?
Improving the performance of how the emulator uses the GPU to process these 2D images is something that shouldn't be too hard for adreno 509,512 ...

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.

Android 10 snapdragon 636 2.2 GHz overclocked 4Gb Ram backend OpenGl

**Comment by Davidcard22** [comment]
![Screenshot_20210113-155005728](https://user-images.githubusercontent.com/77389219/104497126-a5446b00-55b8-11eb-827f-8585628646f8.jpg)

**Comment by sum2012** [comment]
What is your mobile 's model ?

**Comment by Davidcard22** [comment]
ZenFone Max pro m1, snapdragon 636 1.8ghz 4Gb Ram

**Comment by Panderner** [comment]
Here's what happens when I use save state or minimize   the graphical glitch appears.

**Comment by sum2012** [comment]
@Davidcard22 Have you tried latest version in https://buildbot.orphis.net/ppsspp/ ?

**Comment by Ultra24** [comment]
This happened to me on iOS also 
![E2C583F6-65AB-4A5E-903C-91D5A40106CC](https://user-images.githubusercontent.com/62357227/104619208-9a8ee200-565b-11eb-8d85-08d908058798.png)

**Comment by ppmeis** [comment]
Tested on latest Windows build. It's working just fine with default settings:
![image](https://user-images.githubusercontent.com/4381277/104629301-9d102c80-5699-11eb-97fa-0aaa57beb3dc.png)

**Comment by ANR2ME** [comment]
It might be mobile GPU/OpenGL ES related issue

**Comment by ANR2ME** [comment]
> 
> 
> Here's what happens when I use save state or minimize the graphical glitch appears.

@Panderner including on Windows?
Or may be it's mobile GPU/OpenGL ES related issue?

**Comment by Panderner** [comment]
> > Here's what happens when I use save state or minimize the graphical glitch appears.
> 
> @Panderner including on Windows?
> Or may be it's mobile GPU/OpenGL ES related issue?

I've tried OpenGL and Vulkan but it still happens.

**Comment by Panderner** [comment]
> @Panderner including on Windows?

But unable to test windows version because i don't have a PC

**Comment by Panderner** [comment]
Here's what happens when I use save state or minimize PPSSPP the graphical glitch appear.

Without save state or minimize the game works fine
![Screenshot_2021-01-15-13-44-12-329_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/104686317-eca53580-5737-11eb-96a1-c1f2fa51f5cd.jpg)

But when I use save state or minimize the game glitched out.
![Screenshot_2021-01-15-13-44-48-155_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/104686361-06467d00-5738-11eb-9349-06ddeff7edba.jpg)

In order to fix graphical glitches without exiting the game pause and restart mission.

**Comment by ANR2ME** [comment]
Ah you're right, it also happened on Windows, a simple resizing/maximazing the windows will shows graphical glitch.

OpenGL:
![image](https://user-images.githubusercontent.com/7974720/104686460-d4c9b380-572f-11eb-9f7a-000a73ce7be0.png)

Vulkan/D3D11:
![image](https://user-images.githubusercontent.com/7974720/104686372-aba92300-572f-11eb-8639-cd786337375c.png)

**Comment by hrydgard** [comment]
Oh, that is not good. I'll look into it soon.

**Comment by ppmeis** [comment]
I can't reproduce this issue on my computer. Are you using any filtering?

**Comment by ANR2ME** [comment]
> 
> 
> I can't reproduce this issue on my computer. Are you using any filtering?

I'm resetting PPSSPP setting to default and only changed the backend, play the mission and then click maximize button to maximize ppsspp window

**Comment by sum2012** [comment]
@Davidcard22  @Panderner can you test again with latest version in https://buildbot.orphis.net/ppsspp/
 with enable "simulate black transfer effects" ?

**Comment by ANR2ME** [comment]
It's still there on the latest dev build
![image](https://user-images.githubusercontent.com/7974720/104846211-f2ec0b00-590b-11eb-8890-b590f9fe542b.png)

**Comment by sum2012** [comment]
@ANR2ME I think that windows vesion of "play the mission and then click maximize button to maximize ppsspp window" is another issue,

**Comment by ANR2ME** [comment]
may be try fixing it on windows version first, since they have similar symptoms.

**Comment by ANR2ME** [comment]
On Windows, the last working build is v1.10.3-474-gd465ce512
May be @Panderner or @Davidcard22 can confirm whether this is true for Android too?

**Comment by Panderner** [comment]
> On Windows, the last working build is `v1.10.3-474-gd465ce512` `Quick Vulkan crashfix`
> May be @Panderner or @Davidcard22 can confirm whether this is true for Android too?

The issue started in v1.10.3-477-g6d117fb53 and it's true for android too.

**Comment by unknownbrackets** [comment]
That it happens when you minimize implies there's a block transfer or temporary buffer going on that doesn't get saved to RAM.  If #13327 impacted it, that just means we're applying a clear but not some later copy.

-[Unknown]

**Comment by ANR2ME** [comment]
Changing both lines with `gstate.getColorMask() != 0xFFFFFFFF` to `gstate.getColorMask() != 0`  on that commit seems to fix the issue on Metal Slug XX.

**Comment by hrydgard** [comment]
Oh, it's another one of those ... 

I don't think it will help because I suspect it's the "opposite" issue, but try putting the gameid under "DisableFirstFrameReadback" in compat.ini...

**Comment by ANR2ME** [comment]
Adding `ULUS10495 = true` to compat.ini seems to do the trick.

**Comment by Panderner** [comment]
I disabled First Frame Readback for Metal Slug XX and it worked.

**Comment by unknownbrackets** [comment]
Well, this is an awfully unsafe looking safe size:

```
Safe size for 44154000: -64x-16
```

-[Unknown]

**Comment by hrydgard** [comment]
Nice find! 

This should be fixed now after #13941 - please close after testing new builds.

**Comment by Panderner** [comment]
Unfortunately v1.10.3-1596-g48bb797ff is currently only available for 32 bit windows.

**Comment by Panderner** [comment]
As confirm of v1.10.3-1596-g48bb797ff is the issue is now fixed.
